airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestination.java

```diff
@@ -84,70 +85,91 @@ public SQLNamingResolvable getNamingResolver() {
-   * before writing to avoid wasteful record-wise writes.
-   * </p>
-   * <p>
-   * 4. Once all records have been written to buffer, flush the buffer and write any remaining records
-   * to the database (regardless of how few are left).
-   * </p>
-   * <p>
-   * 5. In a single transaction, delete the target tables if they exist and rename the temp tables to
-   * the final table name.
-   * </p>
-   *
-   * @param config - Snowflake-specific configuration object as json.
-   * @param catalog - schema of the incoming messages.
-   * @return consumer that writes singer messages to the database.
-   * @throws Exception - anything could happen!
-   */
   @Override
-  public DestinationConsumer<AirbyteMessage> write(JsonNode config, ConfiguredAirbyteCatalog catalog) throws Exception {
-    // connect to snowflake
-    final Supplier<Connection> connectionFactory = SnowflakeDatabase.getConnectionFactory(config);
-    Map<String, SnowflakeWriteContext> writeBuffers = new HashMap<>();
-    Set<String> schemaSet = new HashSet<>();
-
-    // create temporary tables if they do not exist
-    // we don't use temporary/transient since we want to control the lifecycle
-    for (final ConfiguredAirbyteStream stream : catalog.getStreams()) {
-      final String streamName = stream.getStream().getName();
-      final String schemaName = getNamingResolver().getIdentifier(config.get("schema").asText());
-      final String tableName = getNamingResolver().getRawTableName(streamName);
-      final String tmpTableName = getNamingResolver().getTmpTableName(streamName);
-      if (!schemaSet.contains(schemaName)) {
-        final String query = String.format("CREATE SCHEMA IF NOT EXISTS %s;", schemaName);
-
-        SnowflakeDatabase.executeSync(connectionFactory, query);
-        schemaSet.add(schemaName);
+  protected void connectDatabase(JsonNode config) {
+    connectionFactory = SnowflakeDatabase.getConnectionFactory(config);
+  }
+
+  @Override
+  public void queryDatabase(String query) throws Exception {
+    SnowflakeDatabase.executeSync(connectionFactory, query);
+  }
+
+  @Override
+  public void queryDatabaseInTransaction(String queries) throws Exception {
+    String query = "BEGIN; " + queries + " COMMIT;";
+    SnowflakeDatabase.executeSync(connectionFactory, query, true, rs -> null);
+  }
+
+  @Override
+  public String createRawTableQuery(String schemaName, String streamName) {
+    return String.format(
+        "CREATE TABLE IF NOT EXISTS %s.%s ( \n"
+            + "ab_id VARCHAR PRIMARY KEY,\n"
+            + "\"%s\" VARIANT,\n"
+            + "emitted_at TIMESTAMP WITH TIME ZONE DEFAULT current_timestamp()\n"
+            + ") data_retention_time_in_days = 0;",
+        schemaName, streamName, COLUMN_NAME);
+  }
+
+  @Override
+  public void writeBufferedRecords(int batchSize, CloseableQueue<byte[]> writeBuffer, String schemaName, String tmpTableName) {
```

Comment:
can everything below here just used the default in `AbstractJdbcDestination`?

Reply:
That would remove all the buffering fixes that you introduced for Snowflake that were avoiding the single row inserts, no?